trio/_core/_run.py

```diff
@@ -1151,7 +1151,8 @@ class GuestState:
     run_sync_soon_not_threadsafe = attr.ib()
     done_callback = attr.ib()
     unrolled_run_gen = attr.ib()
-    unrolled_run_next_send = attr.ib(factory=lambda: Value(None), type=object)
+    _value_factory: Callable[[], Value] = lambda: Value(None)
+    unrolled_run_next_send = attr.ib(factory=_value_factory, type=Value)
```

Comment:
I think the type here should be `outcome.Outcome`, not `outcome.Value`? E.g. if `get_events` raises an exception, then `unrolled_run_next_send` will be an `outcome.Error`.

Reply:
Thanks! Fixed.